app/uk/gov/hmrc/agentclientrelationships/model/EnrolmentKey.scala

```diff
@@ -27,12 +27,10 @@ case class EnrolmentKey(service: String, identifiers: Seq[Identifier]) {
   lazy val tag = // note: we intentionally do not use the Identifier's toString below because it uppercases everything!
     s"$service~${identifiers.sorted.map(identifier => s"${identifier.key}~${identifier.value}").mkString("~")}"
   override def toString: String = tag
-  def singleIdentifier: Identifier = // Note: unsafe (i.e. can throw exceptions)
-    if (identifiers.length == 1) identifiers.head else throw new RuntimeException("No single identifier")
-  def singleTaxIdentifier: TaxIdentifier = { // Note: unsafe (i.e. can throw exceptions)
-    val identifier = singleIdentifier
-    ClientIdType.supportedTypes.find(_.enrolmentId == identifier.key).get.createUnderlying(identifier.value)
-  }
```

Comment:
Every single caller of `singleIdentifier` and `singleTaxIdentifier` is now suspect and must be investigated, because by using these methods it shows that it is making the assumption of a single identifier.
The biggest mess is in DeleteRecordRepository (and the other repositories potentially) because there we actually PERSIST data in a format with the built-in assumption that there is only one identifier, and thus when the data is read back it is impossible to reconstruct the original enrolment key. Can we change the storage format?